components/camel-undertow/src/main/java/org/apache/camel/component/undertow/DefaultUndertowHttpBinding.java

```diff
@@ -389,6 +393,63 @@ public Object toHttpResponse(HttpServerExchange httpExchange, Message message) t
         return body;
     }
 
+    /*
+     * when an HTTP response has status code 200 
+     * but does not have content in the body change the status code to 204
+     */
+    private void handleResponseWithNoContent(Exchange camelExchange, HttpServerExchange httpExchange, Object body) {
+        if (body == null) {
+            changeStatusCodeToNoContent(camelExchange, httpExchange);
+        } else {
+            checkBodyForContentAndUpdateStatusCode(camelExchange, httpExchange, body);
+        }
+    }
+    
+    /*
+     * change the status code to no content
+     */
+    private void changeStatusCodeToNoContent(Exchange camelExchange, HttpServerExchange httpExchange) {
+        Message message = camelExchange.getMessage();
+        if (httpExchange.getStatusCode() == 200) {
+            message.getHeaders().put(Exchange.HTTP_RESPONSE_CODE, 204);
+            message.getHeaders().put(Exchange.HTTP_RESPONSE_TEXT, StatusCodes.NO_CONTENT_STRING);
+            httpExchange.setStatusCode(204);
+        }
+    }
+    
+    /*
+     * check body content for being empty
+     */
+    private void checkBodyForContentAndUpdateStatusCode(Exchange camelExchange, HttpServerExchange httpExchange, Object body) {
+        if (body instanceof InputStream) {
+            // reading out the input stream to check it causes issues 
+            // do we want to check using PushbackStream?
+        } else {
+            //
+            TypeConverter tc = camelExchange.getContext().getTypeConverter();
+            ByteBuffer bodyAsByteBuffer = tc.convertTo(ByteBuffer.class, body);
+            if (!bodyAsByteBuffer.hasRemaining()) {
+                this.changeStatusCodeToNoContent(camelExchange, httpExchange);
+            } else {
+                if (hasNoContent(StandardCharsets.UTF_8.decode(bodyAsByteBuffer.asReadOnlyBuffer()).toString())) {
+                    changeStatusCodeToNoContent(camelExchange, httpExchange);
+                }
+            }
+        }
+    }
+    
+    /*
+     * an empty body along with 
+     * various values in the body 
+     * will be treated as no content
+     */
+    private boolean hasNoContent(String bodyAsString) {
+        return (bodyAsString.trim().isEmpty()
+            || bodyAsString.equalsIgnoreCase("No Content")
```

Comment:
@davsclaus I am bit confused, do we really need to check if body has these strings? As I understood from your JIRA https://issues.apache.org/jira/browse/CAMEL-14029, component shouldnt produces such strings at all and instead add the correct status code `204`.

Reply:
this was added based on comments from the other PR (HTTP-Netty)

Reply:
Hey @CodeSmell, sorry for not spotting this earlier. However I guess what @davsclaus meant on the other PR is that, sometimes there might be a component that when camel body is null, they just return hard coded text. However in your case, all what you need to do is, just check if there is an empty/null camel return 204 otherwise 200. It should be very simple